ultra/ultra/evaluate.py

```diff
@@ -138,7 +136,9 @@ def collect_evaluations(evaluation_task_ids: dict):
             episode.eval_train_mode()
 
         episode.info[episode.active_tag] = ray.get(ready_evaluation_task_id)
-        episode.record_tensorboard(recording_step=agent_iteration)
+        episode.record_tensorboard(
+            recording_step=episode.index
+        )  # Record evaluation episodically
```

Comment:
Oh wait okay sorry I just realized that this might not work the way it is currently implemented. Have you tested it by looking at the Tensorboard and results pkls? Since we are running evaluation in parallel, `episode.index` here is most likely not the same episode index when we began evaluation. Therefore, it will record the data at the wrong episode.

Perhaps an easy way to check would be to run evaluation every 5 episodes or something, and then see if the Tensorboard and the results pkls record data at episodes indices that are multiples of 5.